src/HCSR04.cpp

```diff
@@ -20,21 +22,30 @@ double UltraSonicDistanceSensor::measureDistanceCm() {
 }
 
 double UltraSonicDistanceSensor::measureDistanceCm(float temperature) {
+    uint32_t maxDistanceDurationMicroSec;
+
     // Make sure that trigger pin is LOW.
     digitalWrite(triggerPin, LOW);
     delayMicroseconds(2);
     // Hold trigger for 10 microseconds, which is signal for sensor to measure distance.
     digitalWrite(triggerPin, HIGH);
     delayMicroseconds(10);
     digitalWrite(triggerPin, LOW);
+    double speedOfSoundInCmPerMicroSec = 0.03313 + 0.0000606 * temperature; // Cair ≈ (331.3 + 0.606 ⋅ ϑ) m/s
+
+    // Compute max delay based on max distance with 25% margin in microseconds or use absolute timeout
+    maxDistanceDurationMicroSec = 2.5 * maxDistanceCm / speedOfSoundInCmPerMicroSec;
+    if (maxTimeoutMicroSec > 0) {
+    	maxDistanceDurationMicroSec = min(maxDistanceDurationMicroSec, maxTimeoutMicroSec);
```

Comment:
I would not mutate `maxDistanceDurationMicroSec`, it has its meaning and here we set it to something else.
I suggest doing `const uint32_t maxDistanceDurationMicroSec = ...` above (so combining declaration with definition and also making it const and for the result of minimum we create new variable. You can use ternary operator to do if-else logic if you wish, that could be nice.

Reply:
uint32_t or unsigned long uses 4 bytes of memory, this is intended for microcontrollers not for PC with GB of RAM, it's not necessary to consume memory just to be more readable.
I don't think most peoples read libraries code, they just use it.

Reply:
I would use float instead of double, sensors precision is about milimeters when they are excellent, double consumes 8 bytes.
float have 6 decimals that represents 10nm and double have about 14 decimals where we need at very most 2 !

Reply:
@furynick hm I haven't thought about that (the memory)! I am not suggesting this for the people using the library btw., is for us maintainers to have clean code and avoid future mistakes.
The memory -> do we really have to think about it like that? I would expect compiler to optimize the usage of memory in cases like this. Is this a common practice, to try to not create new variables when working with Arduino/embedded code? What is your experience with it?

Float instead of double -> where are we using float? We are using integers, right?

Reply:
Small advice -> don't resolve comments right after you answer them, otherwise I might miss them. In my experience resolve is most useful when used to acknowledge that last comment has been read and the person that has not written it (in this case me) has nothing to add.

Reply:
Thanks for advice, I used them to mark comment resolved after committing related updates.